rest/src/main/java/com/redhat/ipaas/rest/v1/handler/integration/IntegrationHandler.java

```diff
@@ -70,33 +70,29 @@ public String resourceKind() {
 
     @Override
     public Integration create(Integration integration) {
-        Integration requestedIntegration = integration;
-        Optional<String> repoName = requestedIntegration.getGitRepo();
-        if (!repoName.isPresent()) {
-            String generatedRepoName = gitHubService.sanitizeRepoName(integration.getName());
-            requestedIntegration = new Integration.Builder().createFrom(requestedIntegration).gitRepo(generatedRepoName).build();
-        }
 
         String secret = createSecret();
-        ensureGitHubSetup(requestedIntegration, secret);
+        String gitCloneUrl = ensureGitHubSetup(integration, secret);
+
+        Integration requestedIntegration = new Integration.Builder().createFrom(integration).gitRepo(gitCloneUrl).build();
         ensureOpenShiftResources(requestedIntegration, secret);
         return Creator.super.create(requestedIntegration);
     }
 
     // ==========================================================================
 
-    private void ensureGitHubSetup(Integration integration, String secret) {
+    private String ensureGitHubSetup(Integration integration, String secret) {
         try {
             Integration integrationWithGitRepoName = ensureGitRepoName(integration);
-            String repoName = integration.getGitRepo().orElseThrow(() -> new IllegalArgumentException("Missing git repo in integration"));
+            String repoName = integrationWithGitRepoName.getGitRepo().orElseThrow(() -> new IllegalArgumentException("Missing git repo in integration"));
```

Comment:
Just a reminder: We should later be careful what to put into `Integration.gitRepo` as here is supposed the name and not the URL. Also suggest to change `gitRepo` to `gitRepoUrl`. 

For the name we can extract it from the URL anyway (should then be done in `ensureGitRepoName`.

ok for demo-time, will change that then later today.